axelor-human-resource/src/main/java/com/axelor/apps/hr/service/expense/ExpenseServiceImpl.java

```diff
@@ -204,6 +208,64 @@ public ExpenseLine computeAnalyticDistribution(ExpenseLine expenseLine) {
     return expenseLine;
   }
 
+  @Override
+  public ExpenseLine getTotalTaxFromProductAndTotalAmount(ExpenseLine expenseLine) {
```

Comment:
this method is still too complex, please split it. Eventually it would be interesting to create a separate service for this method (for example `ExpenseLineComputationService`)